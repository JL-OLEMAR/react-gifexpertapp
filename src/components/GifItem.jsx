import PropTypes from 'prop-types'

export function GifItem({ title, url }) {
  return (
    <div className='card'>
      <img alt={title} src={url} />
      <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
