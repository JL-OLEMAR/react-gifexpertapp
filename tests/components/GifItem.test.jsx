import { render, screen } from '@testing-library/react'

import { GifItem } from '../../src/components/GifItem.jsx'

describe('Tests in <GifItem />', () => {
  const title = 'Saitama'
  const url = 'https://giphy.com/gifs/yo3TC0yeHd53G'

  test('should match the snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)

    expect(container).toMatchSnapshot()
  })

  test('should show the image with the indicated URL and ALT', () => {
    render(<GifItem title={title} url={url} />)
    const { src, alt } = screen.getByRole('img')

    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('should display the title in the component', () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})
