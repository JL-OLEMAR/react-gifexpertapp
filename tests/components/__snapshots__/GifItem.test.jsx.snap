// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifItem /> should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://giphy.com/gifs/yo3TC0yeHd53G"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
